docs/sdkx-react-native/delegates.mdx

```diff
@@ -0,0 +1,227 @@
+  Admonition,
+  CodeBlock,
+  Tabs,
+  TabItem,
+  LatestSdkVersion,
+  Centered,
+  Image,
+  Intro,
+  SideBySide,
+  DownloadButton,
+  Steps,
+  Step,
+} from "@site/src/components/forDocs";
+
+# Helpshift Delegates {#delegates}
+
+<Intro>
+
+The Helpshift SDK provides delegate callbacks to help app developers track a user's activities within the help section.
+
+</Intro>
+
+## Helpshift Delegate implementation {#delegate-implementation}
+
+- To use Helpshift Delegate please import `helpshiftEmitter`
+```
+import {
+  helpshiftEmitter,
+} from 'helpshift-plugin-sdkx-react-native';
+
+```
+
+```
+const handleCallbacks = () => {
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'receivedUnreadMessageCount') {
+        try {
+          let countData =
+            Platform.OS == 'ios' ? data.eventData : JSON.parse(data.eventData);
+          console.log(countData.count);
+        } catch (err) {
+          console.error('Parsing Error', err);
+        }
+      } else {
+        console.log(JSON.stringify(data));
+      }
+    });
+    helpshiftEmitter.addListener('onUserAuthenticationFailure', (data) => {
+      Toast.show({
+        type: 'error',
+        text1: JSON.stringify(data),
+      });
+    });
+  };
+```
+
+
+## Events {#helpshift-events}
+
+### Helpshift session started event {#session-delegates}
+
+This event gets fired when the Helpshift session starts
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'helpshiftSessionStarted') {
+        //Your code here.
+      }
+    });  
+```
+
+### Helpshift session ended event {#session-delegates}
+
+This event gets fired when the Helpshift session ends.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'helpshiftSessionEnded') {
+        //Your code here.
+      }
+    });  
+```
+
+### Unread Message Count Event {#unreadMessageCount}
+
+This event gets fired when you call `RequestUnreadMessageCount(FetchFromServer)` API to request unread messages count.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'receivedUnreadMessageCount') {
+        //Your code here.
+      } 
+    });
+    helpshiftEmitter.addListener('onUserAuthenticationFailure', (data) => {
+       //Your code here.
+    });
+```
+### Conversation Status Event {#conversation-status}
+
+This event contains information about the current ongoing conversation.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'conversationStatus') {
+        //Your code here.
+        //Event data: `latestIssueId`, `latestIssuePublishId`, `open`
+      }
+    });  
+```
+### Widget Toggle Event {#widget-toggle-event}
+
+This event is triggered when the user opens or exits the chat screen. This event is triggered with a boolean value of `"visible"` key. For your reference, see the below example:
```

Comment:
Add example code for how to read "visible" key in "data"

Reply:
Updated